### PR TITLE
Piotr/tsi charts impl

### PR DESCRIPTION
**Pull request by PiotrKarczmarz**

### CODE REVIEW

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* @import '~tsiclient/tsiclient.css'; */
+
+.chart-rendering-content {
+  -moz-box-sizing: border-box;
+  -webkit-box-sizing: border-box;
+  box-sizing: border-box;
+  height: 500px;
```
Comment (willyt150) [inline Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css]: Think we should try for this component to "fill" the space it's given and have whoever is creating an instance of this component specify its dimensions, **if possible that is**

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+    const aggregateExpressions = [];
+
+    aggregateExpressions.push(
+      new this.client.ux.AggregateExpression(
+        { predicateString: motionPredicate }, // predicate
+        { property: 'Occupied', type: 'Double' }, // measure column
+        ['avg'], // measure type,
+        { from: startDate, to: endDate, bucketSize: '30m' },  // time range
+        null, // split by value, for you probably just null
```
Comment (willyt150) [inline Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts]: We can get rid of `, for you probably just null`

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          self.tryUpdateChart();
+        });
+    } else {
+      self.tokenRetrieved = true;
+      self.tryUpdateChart();
+    }
+  }
+
+  ngOnChanges() {
+    this.tryUpdateChart();
```
Comment (willyt150) [inline Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts]: Should we be utilizing the Angular "differs" to know if there are changes within the collections, or do you think this should catch everything?

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel-brand/hotel-brand.component.ts [hunk 4]
```diff
@@ -71,6 +76,29 @@ export class HotelBrandComponent implements OnInit, OnDestroy {
     });
   }
 
+  private filterSensorIds(hotels: ISpace[]) {
+    for (const hotel of hotels) {
+      const motionSensorIds = this.facilityService.getDescendantSensorIds(hotel.id, 'Motion');
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel-brand/hotel-brand.component.ts]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel-brand/hotel-brand.component.html [hunk 5]
```diff
@@ -13,10 +13,14 @@
 </div>
 
 <div class="hotels-dashboard" *ngIf="hotels !== null">
+  <div>
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel-brand/hotel-brand.component.html]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel/hotel.component.html [hunk 6]
```diff
@@ -13,9 +13,13 @@
   </div>
 </div>
 <div class="floors-dashboard" *ngIf="floors !== null">
+  <div>
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel/hotel.component.html]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel/hotel.component.ts [hunk 7]
```diff
@@ -70,11 +76,36 @@ export class HotelComponent implements OnInit, OnDestroy {
...
+    self.filterSensorIds(floors);
 
     self.subscriptions.push(self.facilityService.getTemperatureAlerts()
       .subscribe(tempAlerts => self.temperatureAlertsUpdated(self.floors, tempAlerts)));
   }
 
+  private filterSensorIds(floors: ISpace[]) {
+
+      for (const floor of floors) {
+      const motionSensorIds = this.facilityService.getDescendantSensorIds(floor.id, 'Motion');
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/hotel/hotel.component.ts]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tenant/tenant.component.html [hunk 8]
```diff
@@ -11,11 +11,18 @@
     </div>
   </div>
 </div>
+
 <div class="hotelbrands-dashboard" *ngIf="hotelBrands !== null">
+  <div>
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tenant/tenant.component.html]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tenant/tenant.component.ts [hunk 9]
```diff
@@ -7,6 +7,7 @@ import { Subscription } from 'rxjs';
 import { ISpaceAlert } from '../services/models/ISpaceAlert';
 import { SubscriptionUtilities } from '../helpers/subscription-utilities';
 import { IPushpinLocation, getPushpinLocation } from '../map/IPushPinLocation';
+import { environment } from 'src/environments/environment.prod';
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tenant/tenant.component.ts]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css [hunk 10]
```diff
@@ -0,0 +1,11 @@
+/* @import '~tsiclient/tsiclient.css'; */
```
[the remark this answers is not in the document]
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts (hunk 2 above)
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts]: Fixed.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css (hunk 1 above)
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.css]: Tried but needs more work to accomplish that. TSI chart "likes" explicit height.

File: Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts (hunk 3 above)
Comment (PiotrKarczmarz) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts]: Good idea - could you point me to example "differes" implementation in the project ?

Comment (willyt150) [reply Source/FacilityManagementWebsite/SmartHotel.FacilityManagementWeb/SmartHotel.FacilityManagementWeb/ClientApp/src/app/tsi-chart/tsi-chart.component.ts]: `map.component.ts` has an example "differs" implementation. I think that's the only one

**Pull request merged**